apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -65,22 +76,24 @@ function PkgModuleViewWrapper({ id, modules }: Props) {
                   )
                   .map(([name]) => name);
 
-    useEffect(() => {
-        const newSearchParams = new URLSearchParams(searchParams);
-        newSearchParams.set('module', selectedModule);
-        setSearchParams(newSearchParams);
-    }, [selectedModule, setSearchParams, searchParams]);
-
     const submitSearch = useCallback(() => {
-        setSelectedModule((prev: string) =>
-            filteredModules.length === 1 ? filteredModules[0] : prev
-        );
-    }, [filteredModules]);
+        if (filteredModules.length === 1) {
+            const convertedSearchParams = new URLSearchParams(searchParams);
```

Comment:
You could also do this with `onChangeModule(filteredModules[0])`, but it's not essential.